### ISSUE TITLE
Collapsable grous state are not saved between renderings

### ISSUE THREAD

**Issue by kimitsu** [mouse0270/module-credits#26]
Hello. It would be a nice improvement to save the states of collapsable groups in Settings Configuration app between renderings (like Tidy UI - Game Settings does). Not only it would improve user experience, it would also (more importantly for me) improve compatibility with modules like Force Client Settings which utilize re-rendering of the app (see https://gitlab.com/kimitsu_desu/force-client-settings/-/issues/7).

**Comment by mouse0270** [comment]
Seems like an easy enough option to add. I will look into it.

**Comment by kimitsu** [comment]
I don't think you have to worry about the search. It's normal to reset the search string and also the collapsable groups when you close and reopen the app.
I've played a bit with the code to see if I can contribute, it goes well, but the "expanding" animation gets replayed at each render. It's a minor visual glitch, maybe not a big deal, but I imagine it may be substandard.

**Comment by mouse0270** [comment]
the animation issue is easy to fix, you simply set the transition to 0s and then remove it after adding the class. I already have a general idea of how to fix it, just want to make sure the user experience is pleasant with things like the animation. lol

**Comment by kimitsu** [comment]
Great idea! Well I trust you have it, then. Thanks!

**Comment by mouse0270** [comment]
This should be resolved in the latest update, if not please re-open this issue

**Comment by kimitsu** [comment]
Thanks! Works like a charm
